router-tests/batch_test.go

```diff
@@ -0,0 +1,385 @@
+						Query: `query employees { employees { id } }`,
+					},
+					{
+						Query: `query employee { employees { isAvailable } }`,
+					},
+				})
+				require.NoError(t, err)
+				require.Equal(t, http.StatusOK, res.Response.StatusCode)
+				entries := getBatchedEntriesForLength(t, res.Body, 2)
+				expected1 := `{"data":{"employees":[{"id":1},{"id":2},{"id":3},{"id":4},{"id":5},{"id":7},{"id":8},{"id":10},{"id":11},{"id":12}]}}`
+				expected2 := `{"data":{"employees":[{"isAvailable":false},{"isAvailable":false},{"isAvailable":false},{"isAvailable":false},{"isAvailable":false},{"isAvailable":false},{"isAvailable":false},{"isAvailable":false},{"isAvailable":false},{"isAvailable":false}]}}`
+				require.Equal(t, expected1, entries[0])
+				require.Equal(t, expected2, entries[1])
+			},
+		)
+	})
+
+	t.Run("verify batching request when batching is not enabled", func(t *testing.T) {
+		t.Parallel()
+
+		testenv.Run(t,
+			&testenv.Config{
+				BatchingConfig: config.BatchingConfig{
+					Enabled:       false,
+					MaxConcurrent: 10,
+				},
+			},
+			func(t *testing.T, xEnv *testenv.Environment) {
+				res, err := xEnv.MakeGraphQLBatchedRequestRequest([]testenv.GraphQLRequest{
+					{
+						Query: `query employees { employees { id } }`,
+					},
+					{
+						Query: `query employee { employees { isAvailable } }`,
+					},
+				})
+				require.NoError(t, err)
+				require.Equal(t, http.StatusBadRequest, res.Response.StatusCode)
+				require.Equal(t, `{"errors":[{"message":"error parsing request body"}]}`, res.Body)
+			},
+		)
+	})
+
+	t.Run("send batch request over max allowed count", func(t *testing.T) {
+		t.Parallel()
+
+		testenv.Run(t,
+			&testenv.Config{
+				BatchingConfig: config.BatchingConfig{
+					Enabled:            true,
+					MaxConcurrent:      10,
+					MaxEntriesPerBatch: 5,
+				},
+			},
+			func(t *testing.T, xEnv *testenv.Environment) {
+				res, err := xEnv.MakeGraphQLBatchedRequestRequest([]testenv.GraphQLRequest{
+					{
+						Query: `query employees { employees { id } }`,
+					},
+					{
+						Query: `query employee { employees { isAvailable } }`,
+					},
+					{
+						Query: `query employee { employees { isAvailable } }`,
+					},
+					{
+						Query: `query employee { employees { isAvailable } }`,
+					},
+					{
+						Query: `query employee { employees { isAvailable } }`,
+					},
+					{
+						Query: `query employee { employees { isAvailable } }`,
+					},
+					{
+						Query: `query employee { employees { isAvailable } }`,
+					},
+				})
+				require.NoError(t, err)
+				require.Equal(t, http.StatusBadRequest, res.Response.StatusCode)
+				require.Equal(t, "unable to process request", res.Body)
+			},
+		)
+	})
+
+	t.Run("verify batched requests when max is unlimited", func(t *testing.T) {
+		t.Parallel()
+
+		testenv.Run(t,
+			&testenv.Config{
+				BatchingConfig: config.BatchingConfig{
+					Enabled:            true,
+					MaxConcurrent:      10,
+					MaxEntriesPerBatch: 0, // 0 means unlimited
```

Comment:
@StarpTech do you think we actually need this? (this is not the default so someone needs to set this, its a big risk for them though, but its the same as setting 1000000 or something)

Reply:
I'd always prefer defensive approach. In that case, a user can simply set it to a higher number if he is not interested in limiting but at least it was explicit.